apps/src/code-studio/header.js

```diff
@@ -146,6 +157,12 @@ header.build = function (scriptData, stageData, progressData, currentLevelId, pu
   }
 };
 
+const ConnectedShareDialog = connect(state => ({
+  isOpen: state.header.shareDialog.isOpen,
+}), dispatch => ({
+  onClose: () => dispatch(hideShareDialog())
+}))(ShareDialog);
```

Comment:
If we're really do need to connect to two different parts of the redux state, my first question would be whether we can find a better way to structure our redux state so that this is not the case. If that isn't possible, I would advocate for having two different connected components (each with their own file) rather than connecting here.